### ISSUE TITLE
Make the web API and related infrastructure FOSS too

### ISSUE THREAD

**Issue by CONIGUERO** [yuzu-emu/yuzu#9783]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

I'm requesting the yuzu web API code, including the multiplayer lobby/room infrastructure, to be released as FOSS just like the emulator itself

### Why would this feature be useful?

Many reasons, which include:

* **Transparency** - People should know how it works and how their game/personal data is handled.
* **Intellectual curiosity** - Same as above but for enriching the community and harbor possible contributions.
* **Self-hosting** - It should be possible to set up a full private multiplayer experience without having to rely on any third-party services.
* **Mantaining the Libre Software Spirit** - Yuzu is a FLOSS emulator licensed under the GPLv2. Having proprietary services attached, even if optional, doesn't bode well with that. Even paid Early-Access users can access the source code of their builds.

**Comment by liamwhite** [comment]
Note: the multiplayer room server source is available here: https://github.com/yuzu-emu/yuzu/blob/master/src/dedicated_room/yuzu_room.cpp

**Comment by CONIGUERO** [comment]
> Note: the multiplayer room server source is available here: https://github.com/yuzu-emu/yuzu/blob/master/src/dedicated_room/yuzu_room.cpp

Yes, thank you, I've seen that. However, that's only for hosting a private "node" in the greater overall nerwork of rooms that the web API brings together. A token and an API url are still mandatory arguments.
